base_layer/core/src/proof_of_work/monero_rx/helpers.rs

```diff
@@ -556,7 +588,21 @@ mod test {
         let mut block_header2 = block_header.clone();
         block_header2.version = 1;
         let hash2 = block_header2.merge_mining_hash();
-        append_merge_mining_tag(&mut block, hash2).unwrap();
+
+        // Try via the API
+        assert!(append_merge_mining_tag(&mut block, hash2).is_err());
+
+        // Now bypass the API
+        let mut extra_field = match ExtraField::try_parse(&block.miner_tx.prefix.extra) {
+            Ok(val) => val,
+            Err(val) => val,
+        };
+        let hash = monero::Hash::from_slice(hash.as_ref());
+        let mut fields = vec![SubField::MergeMining(Some(VarInt(0)), hash)];
+        fields.append(&mut extra_field.0);
+        extra_field.0 = fields;
+        block.miner_tx.prefix.extra = extra_field.into();
+
```

Comment:
can you add some more comments about this, like why is the API failing, and why are we now bypassing the API?
And then do we need to compare them or something?